skimage/filters/_frangi.py

```diff
@@ -0,0 +1,185 @@
+    ----------
+    image : (N, M) ndarray
+        Array with input image data.
+    scale : tuple of floats, optional
+        The range of sigmas used, default [1 10].
+    scale_ratio : float, optional
+        Step size between sigmas, default 2.
+    beta1 : float, optional
+        Frangi correction constant, default 0.5.
+    beta2 : float, optional
+        Frangi correction constant, default 15.
+    black_ridges : boolean, optional
+        Detect black ridges (default) set to true, for
+        white ridges set to false.
+
+    Returns
+    -------
+    out : (N, M) ndarray
+        Filtered image (maximum of pixels across all scales).
+
+    Notes
+    -----
+    Written by Marc Schrijver, 2/11/2001
+    Re-Written by D.Kroon University of Twente (May 2009)
+
+    References
+    ----------
+    .. [1] A. Frangi, W. Niessen, K. Vincken, and M. Viergever. "Multiscale
+           vessel enhancement filtering," In LNCS, vol. 1496, pages 130-137,
+           Germany, 1998. Springer-Verlag.
+    .. [2] Kroon, D.J.: Hessian based frangi vesselness filter.
+    .. [3] http://mplab.ucsd.edu/tutorials/gabor.pdf.
+    """
+
+    from ..feature import hessian_matrix, hessian_matrix_eigvals
+    sigmas = np.arange(scale[0], scale[1], scale_ratio)
+
+    if np.any(np.asarray(sigmas) < 0.0):
+        raise ValueError("Sigma values less than zero are not valid")
+
+    beta1 = 2 * beta1 ** 2
+    beta2 = 2 * beta2 ** 2
+
+    # Make matrices to store all filtered images
+    filtered_list = []
+
+    # Frangi filter for all sigmas
+    for sigma in sigmas:
+
+        # Make 2D hessian
+        (Dxx, Dxy, Dyy) = hessian_matrix(image, sigma)
+
+        # Correct for scale
+        Dxx = (sigma ** 2) * Dxx
+        Dxy = (sigma ** 2) * Dxy
+        Dyy = (sigma ** 2) * Dyy
+
+        # Calculate (abs sorted) eigenvalues and vectors
+        (Lambda1, Lambda2) = hessian_matrix_eigvals(Dxx, Dxy, Dyy)
+
+        # Compute some similarity measures
+        Lambda1[np.where(Lambda1 == 0)] = 1 ** (-10)
+        Rb = (Lambda2 / Lambda1) ** 2
+        S2 = Lambda1 ** 2 + Lambda2 ** 2
+
+        # Compute the output image
+        filtered = np.exp(-Rb / beta1) * (np.ones(np.shape(image)) -
+                                          np.exp(-S2 / beta2))
+
+        if (black_ridges):
+            filtered[np.where(Lambda1 < 0)] = 0
+        else:
+            filtered[np.where(Lambda1 >= 0)] = 0
+
+        # Store the results in 3D matrices
+        filtered_list.append(filtered)
+
+    # Return for every pixel the value of the scale(sigma) with the maximum
+    # output pixel value
+    if len(sigmas) > 1:
+        out = np.maximum.reduce(filtered_list)
+    else:
+        out = filtered
+
+    return out
+
+
+def hessian_filter(image, scale=(1, 10), scale_ratio=2, beta1=0.5, beta2=15):
+    """Returns image filtered with hybrid hessian filter.
+
+    Almost equal to frangi filter, but uses alternative method of smoothing.
+    Can be used to detect edges.
```

Comment:
Actually, I'm concerned about the amount of shared code between these two functions... Do you think you could refactor so that they both call out to a common function, but a flag changes the behaviour of the loop so between one or the other?